Cannot compile on windows following thorvg commit #8866c36

### Godot version

4.0 custom 5bc87025f41566179352ae62478fcb2a141cfdb0

### System information

Windows 10, GTX 1060

### Issue description

Following commit 8866c365829896bf6b4c8ec4be25c161ccaf7282 I think, because it work before.
I cannot compile anymore on Windows

```
scons: Reading SConscript files ...

scons: warning: No version of Visual Studio compiler found - C/C++ compilers most likely not set correctly
File "E:/Developpement/msys64/home/antca/godot/SConstruct", line 3, in <module>
Configuring for Windows: target=release_debug, bits=default
Using MinGW
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
[  5%] Linking Static Library main/libmain.windows.opt.tools.64.a ...
[ 22%] Compiling thirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp ...
=====
b'thirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp: In function \'const char* _simpleXmlSkipXmlEntities(const char*, const char*)\':\nthirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp:104:17: error: \'strncmp\' was not declared in this scope\n  104 |             if (strncmp(itr, xmlEntity[i], xmlEntityLength[i]) == 0) {\n      |                 ^~~~~~~\nthirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp:33:1: note: \'strncmp\' is defined in header \'<cstring>\'; did you forget to \'#include <cstring>\'?\n   32 | #include "tvgXmlParser.h"\n  +++ |+#include <cstring>\n   33 | \nthirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp: In function \'const char* _simpleXmlUnskipXmlEntities(const char*, const char*)\':\nthirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp:122:17: error: \'strncmp\' was not declared in this scope\n  122 |                 strncmp(itr - xmlEntityLength[i], xmlEntity[i], xmlEntityLength[i]) == 0) {\n      |                 ^~~~~~~\nthirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp:122:17: note: \'strncmp\' is defined in header \'<cstring>\'; did you forget to \'#include <cstring>\'?\nthirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp: In function \'bool simpleXmlParseW3CAttribute(const char*, simpleXMLAttributeCb, const void*)\':\nthirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp:461:17: error: \'strlen\' was not declared in this scope\n  461 |     end = buf + strlen(buf);\n      |                 ^~~~~~\nthirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp:461:17: note: \'strlen\' is defined in header \'<cstring>\'; did you forget to \'#include <cstring>\'?\nthirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp:468:28: error: \'strchr\' was not declared in this scope\n  468 |         char* sep = (char*)strchr(buf, \':\');\n      |                            ^~~~~~\nthirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp:468:28: note: \'strchr\' is defined in header \'<cstring>\'; did you forget to \'#include <cstring>\'?\n'
=====
scons: *** [thirdparty/thorvg/src/loaders/svg/tvgXmlParser.windows.opt.tools.64.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:00:05.590]
```


### Steps to reproduce

Git pull
Compile on windows "scons platform=windows use_mingw=yes target=release_debug use_lto=yes"

### Minimal reproduction project

_No response_

No, but different error

```scons: Reading SConscript files ...

scons: warning: No version of Visual Studio compiler found - C/C++ compilers most likely not set correctly
File "E:/Developpement/msys64/home/antca/godot/SConstruct", line 3, in <module>
Configuring for Windows: target=release_debug, bits=default
Using MinGW
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
[ 22%] Compiling thirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp ...
=====
b'thirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp: In function \'std::string svgUtilURLDecode(const char*)\':\nthirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp:218:19: error: \'strlen\' was not declared in this scope\n  218 |     auto length = strlen(src);\n      |                   ^~~~~~\nthirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp:26:1: note: \'strlen\' is defined in header \'<cstring>\'; did you forget to \'#include <cstring>\'?\n   25 | #include "tvgSvgUtil.h"\n  +++ |+#include <cstring>\n   26 | \nthirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp: In function \'std::string svgUtilBase64Decode(const char*)\':\nthirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp:246:19: error: \'strlen\' was not declared in this scope\n  246 |     auto length = strlen(src);\n      |                   ^~~~~~\nthirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp:246:19: note: \'strlen\' is defined in header \'<cstring>\'; did you forget to \'#include <cstring>\'?\n'
=====
scons: *** [thirdparty/thorvg/src/loaders/svg/tvgSvgUtil.windows.opt.tools.64.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:00:05.888]
```

--- a/thirdparty/README.md
+++ b/thirdparty/README.md
@@ -613,6 +613,8 @@ Files extracted from upstream source:
 See `thorvg/update-thorvg.sh` for extraction instructions. Set the version
 number and run the script.
 
+Patches in the `patches` directory should be re-applied after updates.
+
 
 ## vhacd
 

--- a/thirdparty/thorvg/patches/thorvg-pr1159-mingw-fix.patch
+++ b/thirdparty/thorvg/patches/thorvg-pr1159-mingw-fix.patch
@@ -1,0 +1,73 @@
+diff --git a/thirdparty/thorvg/src/loaders/svg/tvgSvgLoader.cpp b/thirdparty/thorvg/src/loaders/svg/tvgSvgLoader.cpp
+index def8ae169a..cf103774c5 100644
+--- a/thirdparty/thorvg/src/loaders/svg/tvgSvgLoader.cpp
++++ b/thirdparty/thorvg/src/loaders/svg/tvgSvgLoader.cpp
+@@ -51,6 +51,7 @@
+ 
+ #define _USE_MATH_DEFINES       //Math Constants are not defined in Standard C/C++.
+ 
++#include <cstring>
+ #include <fstream>
+ #include <float.h>
+ #include <math.h>
+diff --git a/thirdparty/thorvg/src/loaders/svg/tvgSvgPath.cpp b/thirdparty/thorvg/src/loaders/svg/tvgSvgPath.cpp
+index 2b62315de8..32685ee620 100644
+--- a/thirdparty/thorvg/src/loaders/svg/tvgSvgPath.cpp
++++ b/thirdparty/thorvg/src/loaders/svg/tvgSvgPath.cpp
+@@ -50,6 +50,7 @@
+ 
+ #define _USE_MATH_DEFINES       //Math Constants are not defined in Standard C/C++.
+ 
++#include <cstring>
+ #include <math.h>
+ #include <clocale>
+ #include <ctype.h>
+diff --git a/thirdparty/thorvg/src/loaders/svg/tvgSvgSceneBuilder.cpp b/thirdparty/thorvg/src/loaders/svg/tvgSvgSceneBuilder.cpp
+index 8701fe32b1..ae17634f31 100644
+--- a/thirdparty/thorvg/src/loaders/svg/tvgSvgSceneBuilder.cpp
++++ b/thirdparty/thorvg/src/loaders/svg/tvgSvgSceneBuilder.cpp
+@@ -49,6 +49,7 @@
+ */
+ 
+ 
++#include <cstring>
+ #include <string>
+ #include "tvgMath.h"
+ #include "tvgSvgLoaderCommon.h"
+diff --git a/thirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp b/thirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp
+index d5b9cdcf7b..9f269b29a2 100644
+--- a/thirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp
++++ b/thirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp
+@@ -20,6 +20,7 @@
+  * SOFTWARE.
+  */
+ 
++#include <cstring>
+ #include <math.h>
+ #include <memory.h>
+ #include "tvgSvgUtil.h"
+diff --git a/thirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp b/thirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp
+index 2e3d5928d9..1571aa4e25 100644
+--- a/thirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp
++++ b/thirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp
+@@ -20,6 +20,7 @@
+  * SOFTWARE.
+  */
+ 
++#include <cstring>
+ #include <ctype.h>
+ #include <string>
+ 
+diff --git a/thirdparty/thorvg/src/savers/tvg/tvgTvgSaver.cpp b/thirdparty/thorvg/src/savers/tvg/tvgTvgSaver.cpp
+index 9450d80e88..9dd57e5a89 100644
+--- a/thirdparty/thorvg/src/savers/tvg/tvgTvgSaver.cpp
++++ b/thirdparty/thorvg/src/savers/tvg/tvgTvgSaver.cpp
+@@ -24,6 +24,8 @@
+ #include "tvgTvgSaver.h"
+ #include "tvgLzw.h"
+ 
++#include <cstring>
++
+ #ifdef _WIN32
+     #include <malloc.h>
+ #else

--- a/thirdparty/thorvg/src/loaders/svg/tvgSvgLoader.cpp
+++ b/thirdparty/thorvg/src/loaders/svg/tvgSvgLoader.cpp
@@ -51,6 +51,7 @@
 
 #define _USE_MATH_DEFINES       //Math Constants are not defined in Standard C/C++.
 
+#include <cstring>
 #include <fstream>
 #include <float.h>
 #include <math.h>

--- a/thirdparty/thorvg/src/loaders/svg/tvgSvgPath.cpp
+++ b/thirdparty/thorvg/src/loaders/svg/tvgSvgPath.cpp
@@ -50,6 +50,7 @@
 
 #define _USE_MATH_DEFINES       //Math Constants are not defined in Standard C/C++.
 
+#include <cstring>
 #include <math.h>
 #include <clocale>
 #include <ctype.h>

--- a/thirdparty/thorvg/src/loaders/svg/tvgSvgSceneBuilder.cpp
+++ b/thirdparty/thorvg/src/loaders/svg/tvgSvgSceneBuilder.cpp
@@ -49,6 +49,7 @@
 */
 
 
+#include <cstring>
 #include <string>
 #include "tvgMath.h"
 #include "tvgSvgLoaderCommon.h"

--- a/thirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp
+++ b/thirdparty/thorvg/src/loaders/svg/tvgSvgUtil.cpp
@@ -20,6 +20,7 @@
  * SOFTWARE.
  */
 
+#include <cstring>
 #include <math.h>
 #include <memory.h>
 #include "tvgSvgUtil.h"

--- a/thirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp
+++ b/thirdparty/thorvg/src/loaders/svg/tvgXmlParser.cpp
@@ -20,6 +20,7 @@
  * SOFTWARE.
  */
 
+#include <cstring>
 #include <ctype.h>
 #include <string>
 

--- a/thirdparty/thorvg/src/savers/tvg/tvgTvgSaver.cpp
+++ b/thirdparty/thorvg/src/savers/tvg/tvgTvgSaver.cpp
@@ -24,6 +24,8 @@
 #include "tvgTvgSaver.h"
 #include "tvgLzw.h"
 
+#include <cstring>
+
 #ifdef _WIN32
     #include <malloc.h>
 #else

